src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -137,6 +138,14 @@ const WorkspaceContainer = ({
   const [showLockWorkspaceModal, setShowLockWorkspaceModal] = useState(false)
   const [leavingWorkspace, setLeavingWorkspace] = useState(false)
   const workspaceLoaded = !!workspace
+  const isGoogleWorkspaceSyncing = workspaceLoaded && isGoogleWorkspace(workspace) && workspace.workspaceInitialized === false
+
+  useEffect(() => {
+    if (isGoogleWorkspaceSyncing) {
+      Ajax().Metrics.captureEvent(Events.permissionsSynchronizationDelayDisplayed, extractWorkspaceDetails(workspace))
+    }
+    // Only want to event when isGoogleWorkspaceSyncing changes state, not whenever any part of workspace changes.
+  }, [isGoogleWorkspaceSyncing]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
![image](https://user-images.githubusercontent.com/484484/228317054-96ae1413-4fbd-4172-a708-217c67aeef24.png)

Reply:
Ignoring of react-hooks/exhaustive-deps is because it complains about me not declaring `workspace` a dependency. But I'm doing that on purpose to avoid emitting multiple events when other parts of `workspace` changes.

Reply:
ah, I see. So you had to externalize all the checks into a single variable, and then use that as the dependency for useEffect so that it doesn't trigger each time every one of them changes. And then you had to tell `eslint` to just calm down because it was complaining about `workspace`

Reply:
thank you so much. Context switching from Scala to JS/React makes my brain hurt!

Reply:
I know, I'm just waiting for the PR deploy to finish so I can do one more sanity check.

Reply:
Ah gotcha. This whole auto-deploy thing is really nice. I wish we had it for the backend apps!

Reply:
We talked about the functionality as a team, and what we've arrived at with your changes is definitely what we're looking for!

Reply:
Sure, but if the reviewers might be working in terra-ui, they might find it useful learn about `useEffect` and avoiding unintended code executing on each render. TBH, I think this is the first time I caught this just by reading the code… it's taken a long time for this to fully sink into my own brain!

Reply:
Nice catch! I'll have my team take a look at it, but won't gate merging on their re-review